packages/libsql/src/lib/Category.ts

```diff
@@ -252,18 +253,25 @@ class InternalCategory<Event, State, Context>
   }
 
   async reload(streamId: Equinox.StreamId, _requireLeader: boolean, t: TokenAndState<State>) {
-    if (this.access.type === "RollingState") {
-      return this.load(streamId, 0)
+    switch (this.access.type) {
+      // These strategies all have it in common that their state 
+      // can be reconstituted in a single round trip to the database
+      // as such is it as cheap to load them as it is to reload them
+      case "RollingState":
+      case "Snapshot":
+      case "LatestKnownEvent":
+        return this.load(streamId, 0)
+      case "Unoptimized":
+        const streamName = Equinox.StreamName.create(this.categoryName, streamId)
+        const [token, state] = await this.context.loadBatched(
+          streamName,
+          this.codec.decode,
+          this.fold,
+          t.state,
+          Token.version(t.token),
```

Comment:
maybe a helper for getting `[token, state]` from `t` would make things clearer (if there's only one or two never mind; this is obviously not exactly cryptic)